art_internal.hpp

```diff
@@ -31,19 +31,48 @@ class [[nodiscard]] basic_db_leaf_deleter;
 // Internal ART key in binary-comparable format
 template <typename KeyType>
 struct [[nodiscard]] basic_art_key final {
+  // Convert an external key into an internal key supporting
+  // lexicographic comparison.  This is only intended for key types
+  // for which simple conversions are possible.  For complex keys,
+  // including multiple key components or Unicode data, the
+  // application should use a gsl::space<std::byte> which already
+  // supports lexicographic comparison.
   [[nodiscard, gnu::const]] static UNODB_DETAIL_CONSTEXPR_NOT_MSVC KeyType
   make_binary_comparable(KeyType key) noexcept;
 
+  // Convert an internal key into an external key. This is only
+  // intended for key types for which simple conversions are possible.
+  // For complex keys, including multiple key components or Unicode
+  // data, the application should use a gsl::space<std::byte> which
+  // already supports lexicographic comparison.
+  [[nodiscard, gnu::const]] static UNODB_DETAIL_CONSTEXPR_NOT_MSVC KeyType
+  make_external(KeyType key) noexcept;
```

Comment:
Is this used?

Reply:
Yes.
```
  // return the decoded form of the key.
  [[nodiscard, gnu::pure]] constexpr KeyType decode() const noexcept {
    return make_external(key);
  }
```
from
```
../olc_art.hpp:      aleaf->get_key().decode();  // decode the key into the iterator's buffer
../art.hpp:  key_ = leaf->get_key().decode();  // decode key into buffer.
```

Reply:
We should have a conversation about the visitor 'get_key()' returning the internal binary comparable key vs the external key.  To my mind, get_key() returning the external key only works if the inverse transformation is well defined.  It is always well-defined for primitive key types.  It is never well-defined with Unicode components in the key.